### PR TITLE
feat(loki sink): Add `path` option to configure the URL path

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <patrik@ptrk.io>

### CODE REVIEW

File: src/sinks/loki/config.rs [hunk 1]
```diff
@@ -53,16 +53,26 @@ impl Default for CompressionConfigAdapter {
...
 pub struct LokiConfig {
     /// The base URL of the Loki instance.
     ///
-    /// Vector will append `/loki/api/v1/push` to this.
+    /// Vector will append the value of `path` to this.
     pub endpoint: UriSerde,
 
+    /// The path to use in the URL of the Loki instance.
+    ///
+    /// Default to `/loki/api/v1/push`.
```
Comment (spencergilbert) [inline src/sinks/loki/config.rs:72]:
```suggestion
    /// By default, `"/loki/api/v1/push"` is used.
```

This appears to be the style we've used for other notes about default values.

**Review by spencergilbert — CHANGES_REQUESTED** [review]
One small request around the documentation side of the change, otherwise it looks good to me.

I'm curious your particular use case, is there another application listening on a different path with the same API? Or is this for proxying? If you don't mind sharing, of course.

**Comment by Sh4d1** [comment]
Fixed!
> I'm curious your particular use case, is there another application listening on a different path with the same API? Or is this for proxying?

It's for proxying, it's a kinda managed Loki, which happens to not listent on the default URL path :sweat_smile:

**Review by spencergilbert — APPROVED** [review]
Cool - I remember issues with a hardcoded healthcheck path in the past. Thanks for the contribution!

**Comment by Sh4d1** [comment]
Yeah, in my case there isn't any HC endpoint, but it could be interesting to add the HC path as a parameter as well (not really sure it'll be used though haha)

**Comment by spencergilbert** [comment]
> Yeah, in my case there isn't any HC endpoint, but it could be interesting to add the HC path as a parameter as well (not really sure it'll be used though haha)

Yeah probably more elegant than: https://github.com/vectordotdev/vector/blob/master/src/sinks/loki/healthcheck.rs#L24-L29
```rust
        http::StatusCode::NOT_FOUND => {
            debug!("Endpoint `/ready` not found. Retrying healthcheck with top level query.");
            fetch_status("", &config, &client).await?
        }
```

**Pull request merged**